src/dfx-core/src/error/structured_file.rs

```diff
@@ -7,6 +7,9 @@ pub enum StructuredFileError {
     #[error("Failed to parse contents of {0} as json: {1}")]
     DeserializeJsonFileFailed(Box<PathBuf>, serde_json::Error),
 
+    #[error("Failed to parse contents of json: {0}")]
+    DeserializeJsonContentFailed(serde_json::Error),
+
```

Comment:
Please delete this, for two reasons:
1. It won't be referenced once `load_json_string` is deleted
2. This type is `StructuredFileError`.  It is for reporting errors that occur while dealing with a particular file, so every error includes the relevant filename as context.  An error type that is not associated with a particular file would go somewhere else.
```suggestion
```